src/software/backend/simulation/simulator_robot.cpp

```diff
@@ -1,382 +1,441 @@
     return 0.0;
 }
 
-void SimulatorRobotSingleton::kick(float speed_m_per_s)
+Vector SimulatorRobot::velocity()
 {
-    if (auto robot = getCurrentPhysicsRobot().lock())
-    {
-        // TODO: Implement me
-    }
+    return Vector(getVelocityX(), getVelocityY());
 }
 
-void SimulatorRobotSingleton::chip(float distance_m)
+float SimulatorRobot::getVelocityAngular()
 {
-    if (auto robot = getCurrentPhysicsRobot().lock())
+    if (auto robot = physics_robot.lock())
     {
-        // TODO: Implement me
+        return robot->angularVelocity().toRadians();
     }
+    LOG(WARNING) << "SimulatorRobot being used with invalid PhysicsRobot" << std::endl;
+    return 0.0;
 }
 
-void SimulatorRobotSingleton::enableAutokick(float speed_m_per_s)
+float SimulatorRobot::getBatteryVoltage()
 {
-    if (auto robot = getCurrentPhysicsRobot().lock())
-    {
-        // TODO: Implement me
-    }
+    // We currently have 4s batteries on the robot that charge up to a little over
+    // 16V, so we use 16 here to approximate a fully-charged battery
+    // TODO: Should max battery voltage be a constant / injected robot param?
+    // See https://github.com/UBC-Thunderbots/Software/issues/1173
+    return 16.0;
 }
 
-void SimulatorRobotSingleton::enableAutochip(float distance_m)
+void SimulatorRobot::kick(float speed_m_per_s)
 {
-    if (auto robot = getCurrentPhysicsRobot().lock())
+    if (auto robot = physics_robot.lock())
     {
-        // TODO: Implement me
+        if (ball_in_dribbler_area)
+        {
+            Vector kick_vector = Vector::createFromAngle(
+                robot->getRobotWithTimestamp(Timestamp::fromSeconds(0)).orientation());
+            kick_vector = kick_vector.normalize(speed_m_per_s);
+            ball_in_dribbler_area->kick(kick_vector);
+        }
+    }
+    else
+    {
+        LOG(WARNING) << "SimulatorRobot being used with invalid PhysicsRobot"
+                     << std::endl;
     }
 }
 
-void SimulatorRobotSingleton::disableAutokick()
+void SimulatorRobot::chip(float distance_m)
 {
-    if (auto robot = getCurrentPhysicsRobot().lock())
+    if (auto robot = physics_robot.lock())
     {
-        // TODO: Implement me
+        if (ball_in_dribbler_area)
+        {
+            Vector chip_vector = Vector::createFromAngle(
+                robot->getRobotWithTimestamp(Timestamp::fromSeconds(0)).orientation());
+            chip_vector = chip_vector.normalize(distance_m);
+            ball_in_dribbler_area->chip(chip_vector);
+        }
+    }
+    else
+    {
+        LOG(WARNING) << "SimulatorRobot being used with invalid PhysicsRobot"
+                     << std::endl;
     }
 }
 
-void SimulatorRobotSingleton::disableAutochip()
+void SimulatorRobot::enableAutokick(float speed_m_per_s)
 {
-    if (auto robot = getCurrentPhysicsRobot().lock())
-    {
-        // TODO: Implement me
-    }
+    autokick_speed_m_per_s = speed_m_per_s;
+    autochip_distance_m = std::nullopt;
```

Comment:
Could we call `disableAutochip()` here instead? Might as well use the abstractions we already have.

Reply:
done

Reply:
Doesn't look like this change got pushed.

Reply:
pushed now